src/extension/VSProject.cs

```diff
@@ -229,6 +235,74 @@ private void Load()
         }
 
         /// <summary>
+        /// Load a project in the new project format for .NET Core 1.0/1.1. Note that this method
+        /// is only called for file extensions .csproj.
+        /// </summary>
+        /// <returns>True if the project was successfully loaded, false otherwise.</returns>
+        private bool TryLoadDotNetCoreProject() {
+            XmlNode root = _doc.SelectSingleNode("Project");
+
+            if (root != null && SafeAttributeValue(root, "Sdk") != null) 
+            {
+                string targetFramework = _doc.SelectSingleNode("Project/PropertyGroup/TargetFramework").InnerText;
+
+                XmlNode assemblyNameNode = _doc.SelectSingleNode("Project/PropertyGroup/AssemblyName");
+                // Even console apps are dll's even if <OutputType> has value 'EXE'
+                string assemblyName = assemblyNameNode == null ? $"{Name}.dll" : $"{assemblyNameNode.InnerText}.dll";
+
+                XmlNodeList nodes = _doc.SelectNodes("/Project/PropertyGroup");
+
+                string commonOutputPath = null;
+
+                foreach (XmlElement configNode in nodes)
+                {
+                    string configName = GetConfigNameFromCondition(configNode);
+
+                    XmlElement outputPathElement = (XmlElement)configNode.SelectSingleNode("OutputPath");
+                    string outputPath = null;
+                    if (outputPathElement != null)
+                        outputPath = outputPathElement.InnerText;
+
+                    if (configName == null)
+                    {
+                        commonOutputPath = outputPath;
+                        continue;
+                    }
+
+                    if (outputPath == null)
+                        outputPath = commonOutputPath;
+
+                    if (outputPath != null)
+                        _configs.Add(configName, new ProjectConfig(this, configName, outputPath, assemblyName));
+                }
+
+                // By convention there is a Debug and a Release configuration unless others are explicitly 
+                // mentioned in the project file. If we have less than 2 then at least one of those is missing.
+                // We cannot tell however if the existing configuration is meant to replace Debug or Release.
+                // Therefore we just add what is missing. The one that has been replaced will not be used.
```

Comment:
Yes, the new project format makes a lot of implicit assumptions. As a result the files is much slimmer unless you start deviating from the defaults, and even then it only adds what you want to be different compared to the defaults. I think it's kind of consistent with the convention based concepts around for example MVC and other topics.